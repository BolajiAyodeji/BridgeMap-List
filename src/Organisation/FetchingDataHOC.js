import React, { Component } from "react";

const fetchingDataHOC = fetchFunc => Comp => {
  return class extends Component {
    state = {
      data: [],
      isLoading: false,
      error: null
    };

    componentDidMount() {
      this.setState({ isLoading: true });
      fetchFunc(this.props.organisationName || this.props.match.params.name)
        .then(data => this.setState({ data, isLoading: false }))
        .catch(error => this.setState({ error, isLoading: false }));
    }

    render() {
      if (this.state.error) {
        this.props.history.push("/server_error");
      }

      return (
        <div>
          <Comp {...this.props} {...this.state} />
        </div>
      );
    }
  };
};

export default fetchingDataHOC;
