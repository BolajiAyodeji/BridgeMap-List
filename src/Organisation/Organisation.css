.org-info__container {
  margin: 0 auto;
  margin-top: 50px;
  color: #11112f;
  padding: 2rem;
}
.org-info__top {
  display: flex;
  flex-direction: column;
}
.org-info__name,
.org-info__image {
  align-self: center;
}
.org-info__what {
  font-weight: bold;
}
.org-info__image {
  width: 100px;
  height: 100px;
  display: block;
}
.scroll-top {
  width: 50px;
  height: 40px;
  border: 2px solid #7d2d45;
  border-radius: 100%;
  text-align: center;
  position: fixed;
  bottom: 1rem;
  right: 1rem;
}
.scroll-top__button {
  border: none;
  background-color: white;
  border-radius: 100%;
  width: inherit;
  height: inherit;
  outline: none;
  cursor: pointer;
}
.scroll-top__img {
  width: 30px;
  height: 30px;
}

@media screen and (min-width: 600px) {
  .org-info__container {
    width: 70%;
  }
  .org-info__top {
    justify-content: space-between;
    flex-direction: row;
  }
}
