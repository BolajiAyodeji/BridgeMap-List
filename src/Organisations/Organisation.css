.org-info__container{
    margin-left: 1rem;
}
.org-info__what{
    font-weight: bold;
}
.org-info__image{
    width: 200px;
    height: 150px;
}