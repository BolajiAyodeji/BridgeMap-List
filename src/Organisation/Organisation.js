import React, { Fragment } from "react";

import fetchingDataHOC from "./FetchingDataHOC";
import { fetchOrganisation } from "../requests/airtable";
import Loading from "../Shared/Loading/Loading";
import "./Organisation.css";

const Organisation = props => {
  const organisation = props.data || [];
  if (props.isLoading) {
    return (
      <Fragment>
        <Loading />
      </Fragment>
    );
  }
  return (
    <div>
      {organisation.map(info => (
        <div key={info["Name of Organisation"]} className="org-info__container">
          <div className="org-info__top">
            <h2 className="org-info__name">{info["Name of Organisation"]}</h2>
            <img
              className="org-info__image"
              src={info.logo}
              alt="organisation logo"
            />
          </div>
          <p className="org-info__about">{info.About}</p>
          <div className="org-info__general">
            <p className="org-info__type">
              <span className="org-info__what">Type of organisation:</span>
              <span>{info["Type of Organisation"]}</span>
            </p>
            <p className="org-info__reg-number">
              <span className="org-info__what">Registered number:</span>
              <span>{info["Registered number"]}</span>
            </p>
            <p className="org-info__email">
              <span className="org-info__what">Email:</span>
              <span>{info["Organisation email"]}</span>
            </p>
            <p className="org-info__number">
              <span className="org-info__what">Contact number:</span>
              <span>{info["Organisation contact number"]}</span>
            </p>
            <p>
              <span className="org-info__what">Website:</span>
              <a href={info.Website} className="org-info__website">
                {info.Website}
              </a>
            </p>
            <p className="org-info__age">
              <span className="org-info__what"> Age group we work with:</span>
              <span>{info["Age group we work with"]}</span>
            </p>
          </div>
          <p className="org-info__what">Where we are based:</p>
          {info["Where we are based"].map(service => (
            <ul key={service}>
              <li>{service}</li>
            </ul>
          ))}
          <p className="org-info__areas">{info["Areas we work in"]}</p>
          <p className="org-info__what">Services we provide to young people:</p>
          {info["Services Provided to young people"].map(service => (
            <ul key={service}>
              <li>{service}</li>
            </ul>
          ))}
          <p className="org-info__what"> How we work with young people:</p>
          {info["How we work with young people"].map(how => (
            <ul key={how}>
              <li>{how}</li>
            </ul>
          ))}
        </div>
      ))}
    </div>
  );
};

export default fetchingDataHOC(fetchOrganisation)(Organisation);
