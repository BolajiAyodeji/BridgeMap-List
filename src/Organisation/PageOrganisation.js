import React from "react";
import Organisation from "./Organisation";

const PageOrganisation = props => {
  console.log(props);
  return <Organisation organisationName={props.match.params.name} />;
};

export default PageOrganisation;
