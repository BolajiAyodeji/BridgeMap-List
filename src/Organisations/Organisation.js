import React from "react";

import fetchingDataHOC from "./OrganisationsContainer";
import { fetchOrganisation } from "../requests/airtable";
import Header from "../StaticPages/Header";
import "./Organisation.css";

const Organisation = props => {
  const organisation = props.data || [];
  return (
    <div>
      <Header />
      {organisation.map(info => (
        <div key={info["Name of Organisation"]} className="org-info__container">
          <h1 className="org-info__name">{info["Name of Organisation"]}</h1>
          <p className="org-info__what">About</p>
          <p className="org-info__about">{info.About}</p>
          <p className="org-info__what">Website</p>
          <a href={info.Website} className="org-info__website">
            {info.Website}
          </a>
          <p className="org-info__what">Organisation Email </p>
          <p className="org-info__email">{info["Organisation email"]}</p>
          <p className="org-info__what">Organisation Contact Number</p>
          <p className="org-info__number">
            {info["Organisation contact number"]}
          </p>
          <p className="org-info__what">Rgistrated number: </p>
          <p className="org-info__reg-number">{info["Registered number"]}</p>
          <p className="org-info__what">Type of Organisation: </p>
          <p className="org-info__type">{info["Type of Organisation"]}</p>
          <p className="org-info__what">Where we are based:</p>
          {info["Where we are based"].map(service => (
            <ul key={service}>
              <li>{service}</li>
            </ul>
          ))}
          <p className="org-info__areas">{info["Areas we work in"]}</p>
          <p className="org-info__what">Age group we work with: </p>
          <p className="org-info__age">{info["Age group we work with"]}</p>
          <p className="org-info__what">Services we provide to young people:</p>
          {info["Services Provided to young people"].map(service => (
            <ul key={service}>
              <li>{service}</li>
            </ul>
          ))}
          <p className="org-info__what"> How we work with young people:</p>
          {info["How we work with young people"].map(how => (
            <ul key={how}>
              <li>{how}</li>
            </ul>
          ))}
        </div>
      ))}
    </div>
  );
};

export default fetchingDataHOC(fetchOrganisation)(Organisation);
