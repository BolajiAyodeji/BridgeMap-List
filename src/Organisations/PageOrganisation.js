import React from "react";
import Organisation from './Organisation'

const PageOrganisation = (props) => {
    return <Organisation organisationName={props.match.params.name} />
}

export default PageOrganisation