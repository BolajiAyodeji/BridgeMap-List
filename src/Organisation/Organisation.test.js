import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import PageOrganisation from "./PageOrganisation";
import Organisation from "./Organisation";
import FetchingDataHOC from "./FetchingDataHOC";
import fetchingDataHOC from "./FetchingDataHOC";

describe("Page organisation component", () => {
  it("renders without crashing", () => {
    shallow(
      <div>
        <Organisation organisationName="Futureversity" />
      </div>
    );
  });
  it("has a children that has organisationName prop", () => {
    const wrapper = shallow(
      <div>
        <Organisation organisationName="Futureversity" />
      </div>
    );
    expect(
      wrapper.props().children.props.hasOwnProperty("organisationName")
    ).toEqual(true);
  });
});

describe("Fetching data HOC component", () => {
  let WrapperComponent;

  beforeEach(() => {
    class MockComponent extends React.Component {
      render() {
        return <div>Component</div>;
      }
    }
    const mockFetchData = jest.fn();
    mockFetchData.mockReturnValue(Promise.resolve());
    WrapperComponent = fetchingDataHOC(mockFetchData)(MockComponent);
  });

  it("withHoc should render without crashing", () => {
    const wrapper = renderer.create(
      <WrapperComponent organisationName="Futureversity" />
    );

    console.log(wrapper);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
