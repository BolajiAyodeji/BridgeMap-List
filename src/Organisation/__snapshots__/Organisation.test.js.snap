// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetching data HOC component withHoc should render without crashing 1`] = `
<div>
  <div>
    Component
  </div>
</div>
`;
